lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -46,10 +62,59 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def relevant_included_associations_for(serializer, nested_associations = {})
```

Comment:
Yup, let's use the opportunity to get rid of the redundant `serializer` parameters. Basically, every method that is only called on the base serializer should not take it as a parameter, and since we're goign the recursive way, there shouldn't be many places where there's a serializer argument left.